src/McpServerService.cs

```diff
@@ -0,0 +1,192 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using ModelContextProtocol.Attributes;
+using ModelContextProtocol.Server;
+
+namespace ViewAppxPackage
+{
+    /// <summary>
+    /// MCP (Model Context Protocol) Server service that provides tools to interact with AppX package data.
+    /// This service exposes package information through MCP tools, allowing external systems to query
+    /// package family names and other package information.
+    /// 
+    /// Usage:
+    /// To start the application in MCP server mode, use the command line argument:
+    /// view-appxpackage.exe -mcpserver
+    /// 
+    /// In this mode, the application will:
+    /// 1. Load all package data without showing the UI
+    /// 2. Start an MCP server that exposes tools for querying package information
+    /// 3. Wait for package loading to complete before serving requests
+    /// 4. Provide the 'list_package_family_names' tool to retrieve all package family names
+    /// 
+    /// The server will continue running until terminated and can serve multiple MCP tool requests.
+    /// </summary>
+    public class McpServerService
```

Comment:
For this to work we need to be using the `WithToolsFromAssembly` service from the ModelContextProtocol package. To use that we need to switch the app startup to the builder pattern, using Host.CreateApplicationBuilder in the Microsoft.Extensions.Hosting package. And to do that, we need to switch this app from a generated static Main to a coded one, which requires setting the EnableDefaultApplicationEntryPoint project property to false and adding code to App.Xaml.cs (or to a Program.cs)